hooks/pre-commit

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# pre-commit hook for git which builds the docker image for a given cloud-sdk version.
+# Copy into .git/hooks/  folder
```

Comment:
-e comes from plenty of gotchas ([long read](http://mywiki.wooledge.org/BashFAQ/105)) however mostly it'll terminate script with non-zero exit code immediately if any of the functions/commands fail. see: http://redsymbol.net/articles/unofficial-bash-strict-mode/ for similar flags that increase robustness.